core/src/Cuda/Kokkos_Cuda_Task.hpp

```diff
@@ -218,7 +220,7 @@ class TaskQueueSpecialization<SimpleTaskScheduler<Kokkos::Cuda, QueueType>> {
             current_task->as_runnable_task().set_respawn_flag();
           }
 
-          queue.complete((*std::move(current_task)).as_runnable_task(),
+          queue.complete(std::move(*current_task).as_runnable_task(),
```

Comment:
uhm this seems an actual change, i.e. move a ptr vs moving an object? Not sure why the ptr was moved though.

Reply:
This would align with other backends like the Serial backend: https://github.com/kokkos/kokkos/blob/d78c07a3941d6eb0e28ae31c11bdc8de0f2c38fd/core/src/Serial/Kokkos_Serial_Task.hpp#L88-L89

So I guess it is safe

Reply:
The sanitizer complained about the pointer being used after moving and we need to pass a rvalue reference.

Reply:
I am still concerned about this (though I am not sure why one would want to move the pointer in the first place). But this now moves the underlying task not the pointer. As Thomas pointed out the serial backend also only moves the ptr not the task.

Reply:
Note that we populate `current_task` in https://github.com/kokkos/kokkos/blob/949462ea451937951cbdba644d589d7b5a4c2bf8/core/src/Cuda/Kokkos_Cuda_Task.hpp#L123-L127 which also like here is only executed for `warp_lane==0` (and we sync the warp afterwards).

Reply:
but that's a pointer we populate, and you are now moving the thing the pointer points too. If there are any other live pointers to the same task they are now invalidated. I still am not convinced that you can just move `*current_task`.